### PR TITLE
Adding health and readiness probes to Kubernetes documentation

### LINKED ISSUE #399: Document `/health` and `/ready` APIs
Also add detail into Kubernetes section on how to use endpoints for liveness, startup, and readiness probes.

### PR DESCRIPTION
**Pull request by eadgbear**
## 🗣 Description

Adding documentation and samples for health and readiness probes in the Kubernetes documentation.

## 🔨 Related Issues

Closes #399

### CODE REVIEW

File: spiceaidocs/docs/deployment/kubernetes/index.md [hunk 1]
```diff
@@ -98,7 +98,35 @@ helm upgrade --install spiceai spiceai/spiceai --set monitoring.podMonitoring.en
 
 Once the monitoring is enabled, import the [Spice Grafana dashboard](../../clients/grafana/index.md) to visualize the Spice.ai metrics.
 
-### Example values.yaml
+### Health and Readyness
+
+Spice supports three endpoints to help you monitor the state of your Spice: `/health` and `/ready`:
```
Comment (lukekim) [inline spiceaidocs/docs/deployment/kubernetes/index.md]: Avoid the use of "we" "I" "you".

```suggestion
Spice provides two HTTP endpoints to help monitor the state of the Spice runtime: `/health` and `/ready`:
```

File: spiceaidocs/docs/deployment/kubernetes/index.md [hunk 2]
```diff
@@ -98,7 +98,35 @@ helm upgrade --install spiceai spiceai/spiceai --set monitoring.podMonitoring.en
 
 Once the monitoring is enabled, import the [Spice Grafana dashboard](../../clients/grafana/index.md) to visualize the Spice.ai metrics.
 
-### Example values.yaml
+### Health and Readyness
```
Comment (lukekim) [inline spiceaidocs/docs/deployment/kubernetes/index.md]:
```suggestion
### Health and Readiness
```

Also worth providing a link to K8s docs: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes

File: spiceaidocs/docs/deployment/kubernetes/index.md [hunk 3]
```diff
@@ -98,7 +98,35 @@ helm upgrade --install spiceai spiceai/spiceai --set monitoring.podMonitoring.en
...
+```yaml
+livenessProbe:
+  httpGet:
+    path: /health
+    port: 8090
+```
+
+In Kubernetes, this pod will not be marked as *Healthy* until the `/health` endpoint returns `200`.
+
+#### Readyness probe
```
Comment (lukekim) [inline spiceaidocs/docs/deployment/kubernetes/index.md]:
```suggestion
#### Readiness probe
```

**Comment by lukekim** [comment]
Fix title.

File: spiceaidocs/docs/deployment/kubernetes/index.md [hunk 4]
```diff
@@ -98,7 +98,39 @@ helm upgrade --install spiceai spiceai/spiceai --set monitoring.podMonitoring.en
...
+  httpGet:
+    path: /health
+    port: 8090
+```
+
+In Kubernetes, this pod will not be marked as *Healthy* until the `/health` endpoint returns `200`.
+
+#### Readiness probe
+
+The `/ready` endpoint indicates **whether or not the Spice datasets are ready**. This means that while `/health` might indicate that Spice is up and running, until `/ready` reports a status of `200`, your queries may return 0 results
```
Comment (sgrebnov) [inline spiceaidocs/docs/deployment/kubernetes/index.md:120]: It should be `/v1/ready`

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**